posthog/hogql/database/schema/sessions.py

```diff
@@ -0,0 +1,146 @@
+            args=[
+                ast.Call(
+                    name="arrayFlatten",
+                    args=[ast.Call(name="groupArray", args=[ast.Field(chain=[table_name, "urls"])])],
+                )
+            ],
+        ),
+        "entry_url": ast.Call(name="argMinMerge", args=[ast.Field(chain=[table_name, "entry_url"])]),
+        "exit_url": ast.Call(name="argMaxMerge", args=[ast.Field(chain=[table_name, "exit_url"])]),
+        "initial_utm_source": ast.Call(name="argMinMerge", args=[ast.Field(chain=[table_name, "initial_utm_source"])]),
+        "initial_utm_campaign": ast.Call(
+            name="argMinMerge", args=[ast.Field(chain=[table_name, "initial_utm_campaign"])]
+        ),
+        "initial_utm_medium": ast.Call(name="argMinMerge", args=[ast.Field(chain=[table_name, "initial_utm_medium"])]),
+        "initial_utm_term": ast.Call(name="argMinMerge", args=[ast.Field(chain=[table_name, "initial_utm_term"])]),
+        "initial_utm_content": ast.Call(
+            name="argMinMerge", args=[ast.Field(chain=[table_name, "initial_utm_content"])]
+        ),
+        "initial_referring_domain": ast.Call(
+            name="argMinMerge", args=[ast.Field(chain=[table_name, "initial_referring_domain"])]
+        ),
+        "initial_gclid": ast.Call(name="argMinMerge", args=[ast.Field(chain=[table_name, "initial_gclid"])]),
+        "initial_gad_source": ast.Call(name="argMinMerge", args=[ast.Field(chain=[table_name, "initial_gad_source"])]),
+        "event_count_map": ast.Call(
+            name="sumMap",
+            args=[ast.Field(chain=[table_name, "event_count_map"])],
+        ),
+        "pageview_count": ast.Call(name="sum", args=[ast.Field(chain=[table_name, "pageview_count"])]),
+        "autocapture_count": ast.Call(name="sum", args=[ast.Field(chain=[table_name, "autocapture_count"])]),
+        "duration": ast.Call(
+            name="dateDiff",
+            args=[
+                ast.Constant(value="second"),
+                ast.Call(name="min", args=[ast.Field(chain=[table_name, "min_timestamp"])]),
+                ast.Call(name="max", args=[ast.Field(chain=[table_name, "max_timestamp"])]),
+            ],
+        ),
+    }
+
+    select_fields: List[ast.Expr] = []
+    group_by_fields: List[ast.Expr] = [ast.Field(chain=[table_name, "session_id"])]
+
+    for name, chain in requested_fields.items():
+        if name in aggregate_fields:
+            select_fields.append(ast.Alias(alias=name, expr=aggregate_fields[name]))
+        else:
+            select_fields.append(
+                ast.Alias(alias=name, expr=ast.Field(chain=cast(list[str | int], [table_name]) + chain))
+            )
+            group_by_fields.append(ast.Field(chain=cast(list[str | int], [table_name]) + chain))
+
+    return ast.SelectQuery(
+        select=select_fields,
+        select_from=ast.JoinExpr(table=ast.Field(chain=[table_name])),
+        group_by=group_by_fields,
+    )
+
+
+class SessionsTable(LazyTable):
```

Comment:
I'm super interested to watch this PR... 

I couldn't figure out how to make this work properly for session_replay_events (but I'm a HogQL noob and didn't have time to dedicate to it 🤣)